### PR TITLE
fix codeql alerts

### PR DESCRIPTION
**Pull request by ganeshkumarsv**
### What does this PR do?

This PR fixes some of the code scanning alerts.
You can find the PR conversation we had with Agent Core team here: https://github.com/stephengroat/datadog-agent/pull/1/files

This PR fixes the following alerts
https://github.com/DataDog/datadog-agent/pull/9647/checks?check_run_id=4000934041
Feel free to let me know if this PR should be split into multiple small PRs

### Motivation

What inspired you to submit this pull request?
Github Code Scanning

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes
and instructions on how this should be tested in QA.

Describe or link instructions to set up environment 
for testing, if the process requires more than just
running the agent on one of the supported platforms.

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by djmitche — CHANGES_REQUESTED** [review]
The bits I didn't comment on look good 😺

**Review by vickenty — CHANGES_REQUESTED** [review]

**Review by djmitche — APPROVED** [review]
Changes look good.  I think the only thing left is the HTTP responses?

**Review by robertjli — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]
CI looks good and changes on Agent Platform files look good too

File: cmd/system-probe/api/restart.go [hunk 1]
```diff
@@ -27,7 +28,7 @@ func restartModuleHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if target.Name != moduleName {
-		http.Error(w, fmt.Sprintf("invalid module: %s", moduleName), http.StatusBadRequest)
+		http.Error(w, fmt.Sprintf("invalid module: %s", html.EscapeString(string(moduleName))), http.StatusBadRequest)
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply cmd/system-probe/api/restart.go:31]: It is also worth noting that this API is not used in a web/browser context. It is accessed via CLI commands.

File: pkg/dogstatsd/parse_events.go [hunk 2]
```diff
@@ -85,7 +84,9 @@ func parseHeader(rawHeader []byte) (eventHeader, error) {
 
 	// Convert title length to workable type and do a basic validity check on value
 	titleLength, err := parseInt64(rawTitleLength)
-	if err != nil || titleLength < 0 {
+	// Before Go 1.17, we can use the following trick to define MaxInt
+	const MaxInt = ^uint(0) >> 1
+	if err != nil || titleLength < 0 || titleLength > int64(MaxInt) {
```
Comment (L3n41c) [inline pkg/dogstatsd/parse_events.go:89]: As `titleLength` is an `int64`, [there is no possible value that can be greater than `MaxInt`](https://play.golang.org/p/I4wtFzyNunN).
So, `titleLength > int64(MaxInt)` is an antilogy, isn’t it ?

Comment (L3n41c) [inline pkg/dogstatsd/parse_events.go:89]: As `titleLength` is an `int64`, [there is no possible value that can be greater than `MaxInt`](https://play.golang.org/p/I4wtFzyNunN).
So, `titleLength > int64(MaxInt)` is an antilogy, isn’t it ?

File: pkg/dogstatsd/parse_events.go [hunk 3]
```diff
@@ -96,7 +97,7 @@ func parseHeader(rawHeader []byte) (eventHeader, error) {
 
 	// Convert text length to workable type and do a basic validity check on value
 	textLength, err := parseInt64(rawTextLength)
-	if err != nil || textLength < 0 {
+	if err != nil || textLength < 0 || textLength > int64(MaxInt) {
```
Comment (L3n41c) [inline pkg/dogstatsd/parse_events.go:100]: As above, I think that `textLength > int64(MaxInt)` can never be satisfied.

File: pkg/dogstatsd/parse_events.go (hunk 2 above)
Comment (djmitche) [reply pkg/dogstatsd/parse_events.go:89]: On 64-bit machines (including that used for the playground), you are correct.  In a 32-bit context, int is the same as int32, so there are lots of int64 values that are greater than MaxInt.

This is one of those holdovers in Go from the C/C++ world that has caused lots of bugs in C/C++ and continues to do so in Go.

**Review by brycekahle — APPROVED** [review]

File: pkg/dogstatsd/parse_events.go (hunk 2 above)
Comment (vickenty) [reply pkg/dogstatsd/parse_events.go:89]: `parseInt64` is only used in this function. So I think we can change `parseInt64() int64` to `parseUint() uint` (adjusting implementation accordingly) and avoid the need for range checks here. WDYT?

**Review by vickenty — APPROVED** [review]

**Review by gmarabout — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]
lgmt for snmp related changes

**Comment by djmitche** [comment]
I pinged some slack channels for the remaining two teams.  It'd be great to get this merged for 7.33.0, so before Friday.

**Review by paulcacheux — APPROVED** [review]

**Comment by djmitche** [comment]
The APM-related changes (pkg/trace) _look_ fine to me, but I don't know where the skeletons are buried in that package.  So, I've added the APM team to the QA list (by adding a label), and they can have a look and test it out in the QA process.

**Pull request merged**